samplesheets/tests/test_views_api_taskflow.py

```diff
@@ -250,6 +348,537 @@ def test_post_created(self):
+
+        with self.login(self.user_contrib):
+            response = self.client.put(self.url, {'path': self.coll.path})
+
+        self.assertEqual(response.status_code, 400)
+        expected = [
+            '{} {}'.format(
+                'Updating ' + IRODS_TICKET_EX_MSG + ':',
+                IRODS_TICKET_NO_UPDATE_FIELDS_MSG + ' path',
+            )
+        ]
+        self.assertEqual(response.json(), expected)
+        self.assertEqual(self.get_tl_event_count('update'), 0)
+        self.assertEqual(self.get_app_alert_count('update'), 0)
+
+    def test_update_invalid_date(self):
+        """Test put() in IrodsAccessTicketUpdateAPIView with invalid date"""
+        self.assertEqual(IrodsAccessTicket.objects.count(), 1)
+        self.assertEqual(self.get_tl_event_count('update'), 0)
+        self.assertEqual(self.get_app_alert_count('update'), 0)
+        invalid_date = 'invalid'
+        with self.login(self.user_contrib):
+            response = self.client.put(self.url, {'date_expires': invalid_date})
+
+        self.assertEqual(response.status_code, 400)
+        expected = "Updating {}: Invalid data: ".format(IRODS_TICKET_EX_MSG)
+        self.assertIn(expected, str(response.json()))
+        self.assertEqual(self.get_tl_event_count('update'), 0)
+        self.assertEqual(self.get_app_alert_count('update'), 0)
+
+
+class TestIrodsAccessTicketDestroyAPIView(TestIrodsAccessTicketAPIViewBase):
+    """Tests for IrodsAccessTicketDeleteAPIView"""
+
+    def setUp(self):
+        super().setUp()
+        self.ticket_str = TICKET_STR
+        self.label = TICKET_LABEL
+        self.date_expires = (
+            timezone.localtime() + timedelta(days=1)
+        ).isoformat()
+        # Create ticket in database and iRODS
+        self.ticket = self.make_irods_ticket(
+            study=self.study,
+            assay=self.assay,
+            path=self.coll.path,
+            user=self.user,
+            ticket=self.ticket_str,
+            label=self.label,
+            date_expires=timezone.localtime() + timedelta(days=1),
+        )
+        self.irods_backend.issue_ticket(
+            self.irods,
+            'read',
+            self.coll.path,
+            ticket_str=self.ticket_str,
+            expiry_date=None,
+        )
+        self.url = reverse(
+            'samplesheets:api_irods_ticket_delete',
+            kwargs={'irodsaccessticket': self.ticket.sodar_uuid},
+        )
+
+    def test_delete(self):
+        """Test delete() in IrodsAccessTicketDeleteAPIView"""
+        self.assertEqual(IrodsAccessTicket.objects.count(), 1)
+        self.assertEqual(self.get_tl_event_count('delete'), 0)
+        self.assertEqual(self.get_app_alert_count('delete'), 0)
+
+        with self.login(self.user_contrib):
+            response = self.client.delete(self.url)
+
+        self.assertEqual(response.status_code, 204)
+        expected = {'detail': IRODS_TICKET_DELETED_MSG}
```

Comment:
Here we also shouldn't care about the message and the view should be returning the DRF default anyway. Deleted means deleted.